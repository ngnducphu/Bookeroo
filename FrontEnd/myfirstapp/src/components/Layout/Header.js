import React, { useEffect } from "react";
import { Link } from "react-router-dom";
import { connect } from "react-redux";
import { searchBooks } from "../../actions/bookActions";
import { syncLocalStorage } from "../../actions/cartActions";
import { useHistory, useLocation } from "react-router";
import { Badge } from "reactstrap";
const Header = (props) => {
  const location = useLocation();

  useEffect(() => {
    props.syncLocalStorage();
    // eslint-disable-next-line react-hooks/exhaustive-deps
  }, []);
  const history = useHistory();
  const handleOptionSelect = (e) => {
    e.preventDefault();
    const searchTerm = e.currentTarget.text;
    props.searchBooks("category", searchTerm, history);
  };

  return (
    <div>
      <nav className="navbar navbar-expand-sm navbar-dark bg-primary mb-4">
        <div className="container">
          <Link className="navbar-brand" to="/main">
            Bookeroo
          </Link>
          <button
            className="navbar-toggler"
            type="button"
            data-toggle="collapse"
            data-target="#mobile-nav"
          >
            <span className="navbar-toggler-icon" />
          </button>

          <div className="collapse navbar-collapse" id="mobile-nav">
            <ul className="navbar-nav mr-auto">
              <li className="nav-item">
                <Link className="nav-link" to="/aboutus">
                  About us
                </Link>
              </li>
              <li className="nav-item">
                <Link className="nav-link" to="/contactus">
                  Contact us
                </Link>
              </li>
              {location.pathname === "/main" ? (
                <li className="nav-item">
                  <div className="dropdown">
                    <a
                      className="nav-link dropdown-toggle"
                      role="button"
                      id="dropdownMenuLink"
                      data-toggle="dropdown"
                      aria-haspopup="true"
                      aria-expanded="false"
                      href="https://www.bookdepository.com/"
                      onClick={(e) => {
                        e.currentTarget.click();
                      }}
                    >
                      Category
                    </a>

                    <div
                      className="dropdown-menu"
                      aria-labelledby="dropdownMenuLink"
                    >
                      <a
                        className="dropdown-item"
                        onClick={handleOptionSelect}
                        href="https://www.bookdepository.com/"
                      >
                        Science Fiction
                      </a>
                      <div className="dropdown-divider"></div>
                      <a
                        href="https://www.bookdepository.com/"
                        className="dropdown-item"
                        onClick={handleOptionSelect}
                      >
                        Novel
                      </a>

                      <div className="dropdown-divider"></div>
                      <a
                        href="https://www.bookdepository.com/"
                        className="dropdown-item"
                        onClick={handleOptionSelect}
                      >
                        Fantasy
                      </a>
                      <div className="dropdown-divider"></div>

                      <a
                        href="https://www.bookdepository.com/"
                        className="dropdown-item"
                        onClick={handleOptionSelect}
                      >
                        Thriller
                      </a>
                      <div className="dropdown-divider"></div>

                      <a
                        href="https://www.bookdepository.com/"
                        className="dropdown-item"
                        onClick={handleOptionSelect}
                      >
                        Romance
                      </a>
                    </div>
                  </div>
                </li>
              ) : (
                ""
              )}
            </ul>
            {props.security.validToken !== true ? (
              <ul className="navbar-nav ml-auto">
                <li className="nav-item">
                  <Link className="nav-link " to="/register">
                    Sign Up
                  </Link>
                </li>
                <li className="nav-item">
                  <Link className="nav-link" to="/login">
                    Login
                  </Link>
                </li>
              </ul>
            ) : (
              <ul className="navbar-nav ml-auto">
                <li className="nav-item">
                  <Link className="nav-link " to="/shoppingcart">
                    <i className="fa fa-shopping-cart ">
                      <Badge variant="info">{props.cart.length} </Badge>
                    </i>
                  </Link>
                </li>
                &nbsp;&nbsp;&nbsp;&nbsp;
                <li className="nav-item">
                  <Link className="nav-link " to="/profile">
                    My Profile
                  </Link>
                </li>
                {props.security.user.role === "ROLE_ADMIN" ? (
                  <li className="nav-item">
                    <Link className="nav-link " to="/admin">
                      Admin
                    </Link>
                  </li>
                ) : (
                  ""
                )}
              </ul>
            )}
          </div>
        </div>
      </nav>
    </div>
  );
};

const mapStateToProps = (state) => ({
  security: state.security,
  cart: state.cartState.cart,
});
export default connect(mapStateToProps, { searchBooks, syncLocalStorage })(
  Header
);

export const HeaderPlain = () => {
  return (
    <div>
      <nav className="navbar navbar-expand-sm navbar-dark bg-primary mb-4">
        <div className="container">
          <Link className="navbar-brand" to="/main">
            Bookeroo
          </Link>
          <button
            className="navbar-toggler"
            type="button"
            data-toggle="collapse"
            data-target="#mobile-nav"
          >
            <span className="navbar-toggler-icon" />
          </button>

          <div className="collapse navbar-collapse" id="mobile-nav">
            <ul className="navbar-nav mr-auto">
              <li className="nav-item">
                <Link className="nav-link" to="/aboutus">
                  About us
                </Link>
              </li>
              <li className="nav-item">
                <Link className="nav-link" to="/contactus">
                  Contact us
                </Link>
              </li>
              {"testing" === "/main" ? (
                <li className="nav-item">
                  <div className="dropdown">
                    <a
                      className="nav-link dropdown-toggle"
                      role="button"
                      id="dropdownMenuLink"
                      data-toggle="dropdown"
                      aria-haspopup="true"
                      aria-expanded="false"
                      href="https://www.bookdepository.com/"
                      onClick={(e) => {
                        e.currentTarget.click();
                      }}
                    >
                      Category
                    </a>

                    <div
                      className="dropdown-menu"
                      aria-labelledby="dropdownMenuLink"
                    >
                      <a
                        className="dropdown-item"
                        href="https://www.bookdepository.com/"
                      >
                        Science Fiction
                      </a>
                      <div className="dropdown-divider"></div>
                      <a
                        href="https://www.bookdepository.com/"
                        className="dropdown-item"
                      >
                        Novel
                      </a>

                      <div className="dropdown-divider"></div>
                      <a
                        href="https://www.bookdepository.com/"
                        className="dropdown-item"
                      >
                        Fantasy
                      </a>
                      <div className="dropdown-divider"></div>

                      <a
                        href="https://www.bookdepository.com/"
                        className="dropdown-item"
                      >
                        Thriller
                      </a>
                      <div className="dropdown-divider"></div>

                      <a
                        href="https://www.bookdepository.com/"
                        className="dropdown-item"
                      >
                        Romance
                      </a>
                    </div>
                  </div>
                </li>
              ) : (
                ""
              )}
            </ul>

            <ul className="navbar-nav ml-auto">
              <li className="nav-item">
                <Link className="nav-link " to="/profile">
                  My Profile
                </Link>
              </li>
            </ul>
          </div>
        </div>
      </nav>
    </div>
  );
};
