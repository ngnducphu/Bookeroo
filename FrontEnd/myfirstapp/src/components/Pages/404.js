import React from "react";
import "./404.css";
const PageNotFound = () => {
  return (
    <div id="notfound" data-testid="notfound">
      <div className="notfound">
        <div className="notfound-404">
          <h1>
            <span>4</span>
            <span>0</span>
            <span>4</span>
          </h1>
        </div>
        <h2>We are sorry, page not found </h2>
      </div>
    </div>
  );
};

export default PageNotFound;
