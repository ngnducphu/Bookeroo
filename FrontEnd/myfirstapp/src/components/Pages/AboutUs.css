.about-us-text {
  font-weight: bold;
  font-size: 50px;
}

.title-content-text {
  font-weight: bold;
  font-size: 25px;
}
