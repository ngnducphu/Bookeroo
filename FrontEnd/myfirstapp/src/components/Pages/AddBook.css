p.pad {
  /* border: 5px solid white; */
  padding-bottom: 5%;
}
