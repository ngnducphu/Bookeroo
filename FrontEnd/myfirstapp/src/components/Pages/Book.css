.containerr {
  position: absolute;
  top: 10%;
  left: 23%;
  width: 50%;
}

.text {
  display: inline;
  width: 100%;
}

.read-or-hide {
  color: rgb(192, 192, 192);
  cursor: pointer;
}

p.ex1 {
  /* border: 5px solid white; */
  padding-bottom: 5%;
}

img.rresize {
  float: left;
  width: 400px;
  height: 620px;
}

.btn-add-to-cart {
  margin-top: 5%;
}
