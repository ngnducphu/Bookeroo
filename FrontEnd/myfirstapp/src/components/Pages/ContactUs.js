import React from "react";
import "./ContactUs.css";
const ContactUs = () => {
  return (
    <div className="container">
      <div className="row">
        <div className="col-6">
          <p className="contact-us">Contact Us</p>
          <p>
            At Bookeroo, we love to talk to our customers! We're here to answer
            all your book-related inquiries (we also tell some great jokes). Our
            goal is to provide you with an stress-free shopping experience.
            We've got an exceptionally savvy, energetic Book Concierge team here
            at our Melbourne ready to get you all the answers you're looking
            for.
          </p>
        </div>
        <div className="col-6">
          <form>
            <div className="form-group">
              <label htmlFor="fullname">Fullname</label>
              <input
                type="text"
                className="form-control"
                id="fullname"
                aria-describedby="fullname"
                placeholder="Enter fullname"
              />
            </div>
            <div className="form-group">
              <label htmlFor="email">Email address</label>
              <input
                type="email"
                className="form-control"
                id="email"
                aria-describedby="emailHelp"
                placeholder="Enter email"
              />
            </div>
            <div className="form-group">
              <label htmlFor="email">Text</label>
              <textarea
                className="form-control"
                id="email"
                aria-describedby="emailHelp"
                placeholder="Enter your enquiry here"
              ></textarea>
            </div>

            <button type="submit" className="btn btn-primary">
              SEND
            </button>
          </form>
        </div>
      </div>
    </div>
  );
};

export default ContactUs;

export const ContactusPlain = (props) => {
  return (
    <div className="container">
      <div className="row">
        <div className="col-6">
          <p className="contact-us">Contact Us</p>
          <div data-testid="description">
            <p>
              At Bookeroo, we love to talk to our customers! We're here to
              answer all your book-related inquiries (we also tell some great
              jokes). Our goal is to provide you with an stress-free shopping
              experience. We've got an exceptionally savvy, energetic Book
              Concierge team here at our Melbourne ready to get you all the
              answers you're looking for.
            </p>
          </div>
        </div>
        <div className="col-6">
          <form>
            <div className="form-group">
              <label htmlFor="fullname">Fullname</label>
              <input
                type="text"
                className="form-control"
                id="fullname"
                aria-describedby="fullname"
                placeholder="Enter fullname"
              />
            </div>
            <div className="form-group">
              <label htmlFor="email">Email address</label>
              <input
                type="email"
                className="form-control"
                id="email"
                aria-describedby="emailHelp"
                placeholder="Enter email"
              />
            </div>
            <div className="form-group">
              <label htmlFor="email">Text</label>
              <textarea
                className="form-control"
                id="email"
                aria-describedby="emailHelp"
                placeholder="Enter your enquiry here"
              ></textarea>
            </div>
            <div data-testid="button">
              {/* {label} */}
              <button type="submit" className="btn btn-primary">
                SEND
              </button>
            </div>
          </form>
          <div data-testid="row-err" className="row err"></div>
        </div>
      </div>
    </div>
  );
};
