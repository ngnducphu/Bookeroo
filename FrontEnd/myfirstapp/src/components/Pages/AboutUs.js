import React from "react";
import "./AboutUs.css";
/* A simple aboutus page */
const AboutUs = () => {
  return (
    <div className="container">
      <p className="text-center about-us-text"> About Us</p>
      <div className="row">
        <div className="col-6">
          <a href="https://ibb.co/SQwF7fK">
            <img
              src="https://i.ibb.co/yhFHB48/book-search.png"
              alt="book-search"
              border="0"
            />
          </a>{" "}
        </div>
        <div className="col-6">
          <p className="title-content-text">
            Redefining book selling/buying experience
          </p>
          At Bookeroo we want to democratize knowlegde by providing an
          accessible way for people to sell and buy book online. We believe in
          the power of knowledge and community.{" "}
        </div>
      </div>
    </div>
  );
};

export default AboutUs;

export const AboutusPlain = (props) => {
  return (
    <div className="container">
      <p className="text-center about-us-text"> About Us</p>
      <div className="row">
        <div className="col-6">
          <a href="https://ibb.co/SQwF7fK">
            <img
              src="https://i.ibb.co/yhFHB48/book-search.png"
              alt="book-search"
              border="0"
            />
          </a>{" "}
        </div>
        <div data-testid="description" className="col-6">
          <p className="title-content-text">
            Redefining book selling/buying experience
          </p>
          At Bookeroo we want to democratize knowlegde by providing an
          accessible way for people to sell and buy book online. We believe in
          the power of knowledge and community.{" "}
        </div>
      </div>
    </div>
  );
};
