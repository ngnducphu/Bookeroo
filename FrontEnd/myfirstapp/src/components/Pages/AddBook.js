import React, { useState } from "react";
import "./AddBook.css";
import { addBook } from "../../actions/bookActions";
import { connect } from "react-redux";
import { useHistory } from "react-router";
import { ToastContainer } from "react-toastify";
import "react-toastify/dist/ReactToastify.css";

const AddBook = (props) => {
  const [bookAdd, setBookAdd] = useState({
    book: {
      userId: "",
      title: "",
      isbn10: "",
      isbn13: "",
      longDescription: "",
      shortDescription: "",
      publishedDate: "",
      coverImageUrl: "",
      status: "USED",
      price: 0,
      quantity: 0,
    },
    authorNames: [""],
    categoryNames: ["Science Fiction"],
    publisherName: "",
  });
  // book state object

  const history = useHistory();
  const handleAddBook = (e) => {
    e.preventDefault();
    window.scrollTo(0, 0);

    console.log(props.person);
    const userID = parseInt(props.id);
    const newBookAdd = { ...bookAdd };

    newBookAdd.book.userId = userID;
    setBookAdd(newBookAdd);

    props.addBook(bookAdd, history);
  };
  return (
    <div className="row">
      <div data-testid="container" className="container">
        <ToastContainer position="top-center" autoClose={2000} />

        <div className="column">
          <h1 data-testid="title">Sell A Book Form</h1>
          <h5 data-testid="form-description">
            Please fill in the form below with the details of the book you want
            to sell. Thank you for choosing our platform to sell your book!
          </h5>
          <hr></hr>
          <br></br>
          <form>
            <div className="form-group">
              <i className="fas fa-book-open"> &nbsp;&nbsp;</i>
              <label data-testid="book-title" htmlFor="Book Title">
                {" "}
                Book Title
              </label>
              <input
                type="text"
                className="form-control"
                id="Book Title"
                aria-describedby="Book Title"
                placeholder="Enter Book Title"
                onChange={(e) => {
                  const newBookAdd = { ...bookAdd };
                  newBookAdd.book.title = e.target.value;
                  // assigning the value entered by the user as the title of the new book
                  setBookAdd(newBookAdd);
                }}
              />
            </div>
            <div className="form-group">
              <i className="fas fa-user"></i> &nbsp;&nbsp;{" "}
              <label htmlFor="author">Author</label>
              <input
                type="text"
                className="form-control"
                id="Author"
                aria-describedby="Author"
                placeholder="Enter Author Name"
                onChange={(e) => {
                  const newBookAdd = { ...bookAdd };
                  const newAuthorNames = [];
                  newAuthorNames.push(e.target.value);
                  newBookAdd.authorNames = newAuthorNames;
                  setBookAdd(newBookAdd);
                }}
              />
            </div>
            <div className="form-group">
              <i className="fa fa-list-alt" aria-hidden="true"></i>&nbsp;&nbsp;
              <label htmlFor="category">Category</label>
              <select
                name="categorytype"
                className="form-control form-control-md"
                value={bookAdd.categoryNames[0]}
                onChange={(e) => {
                  const newCategory = [];
                  newCategory.push(e.target.value);
                  const newBookAdd = { ...bookAdd };
                  newBookAdd.categoryNames = newCategory;
                  setBookAdd(newBookAdd);
                }}
              >
                <option defaultValue value="Science Fiction">
                  {" "}
                  Science Fiction
                </option>
                <option value="Novel">Novel</option>
                <option value="Fantasy">Fantasy</option>
                <option value="Thriller">Thriller</option>
                <option value="Romance">Romance</option>
              </select>
            </div>
            <div className="form-group">
              <i className="fas fa-info-circle"></i> &nbsp;&nbsp;{" "}
              <label htmlFor="status">Status (Published or Not)</label>
              <select
                className="form-control form-control-md"
                onChange={(e) => {
                  const newBookAdd = { ...bookAdd };
                  newBookAdd.book.status = e.target.value;
                  setBookAdd(newBookAdd);
                }}
              >
                {" "}
                <option defaultValue value="USED">
                  {" "}
                  Used
                </option>
                <option value="NEW">New</option>
              </select>
            </div>
            <div className="form-group">
              <i className="fas fa-dollar-sign"></i> &nbsp;&nbsp;{" "}
              <label htmlFor="price">Price</label>
              <input
                type="text"
                className="form-control"
                id="price"
                aria-describedby="price"
                placeholder="Enter Price"
                onChange={(e) => {
                  const newBookAdd = { ...bookAdd };
                  newBookAdd.book.price = parseFloat(e.target.value);
                  setBookAdd(newBookAdd);
                }}
              />
            </div>
            <div className="form-group">
              <i className="fas fa-info-circle"></i> &nbsp;&nbsp;{" "}
              <label htmlFor="quantity">Quantity</label>
              <input
                type="text"
                className="form-control"
                id="quantity"
                aria-describedby="quantity"
                placeholder="Enter Quantity"
                onChange={(e) => {
                  const newBookAdd = { ...bookAdd };
                  newBookAdd.book.quantity = parseInt(e.target.value);
                  setBookAdd(newBookAdd);
                }}
              />
            </div>
            <div className="form-group">
              <i className="fas fa-info-circle"></i> &nbsp;&nbsp;{" "}
              <label htmlFor="quantity">Publisher</label>
              <input
                type="text"
                className="form-control"
                id="publisher-name"
                aria-describedby="publisher-name"
                placeholder="Enter Publisher"
                onChange={(e) => {
                  const newBookAdd = { ...bookAdd };
                  newBookAdd.publisherName = e.target.value;
                  setBookAdd(newBookAdd);
                }}
              />
            </div>
            <div className="form-group">
              <i className="fas fa-bars"></i> &nbsp;&nbsp;{" "}
              <label htmlFor="long description">Long Description</label>
              <textarea
                className="form-control"
                id="long description"
                aria-describedby="long description"
                placeholder="Enter a long description"
                onChange={(e) => {
                  const newBookAdd = { ...bookAdd };
                  newBookAdd.book.longDescription = e.target.value;
                  setBookAdd(newBookAdd);
                }}
              ></textarea>
            </div>
            <div className="form-group">
              <i className="fas fa-bars"></i> &nbsp;&nbsp;{" "}
              <label htmlFor="short description">Short Description</label>
              <textarea
                className="form-control"
                id="short description"
                aria-describedby="short description"
                placeholder="Enter a short description"
                onChange={(e) => {
                  const newBookAdd = { ...bookAdd };
                  newBookAdd.book.shortDescription = e.target.value;
                  setBookAdd(newBookAdd);
                }}
              ></textarea>
            </div>
            <div className="form-group">
              &nbsp;&nbsp; <i className="fas fa-barcode"></i> &nbsp;&nbsp;{" "}
              <label htmlFor="info">ISBN 10</label>
              <input
                type="text"
                className="form-control"
                id="ISBN-10"
                aria-describedby="ISBN"
                placeholder="Enter ISBN 10"
                onChange={(e) => {
                  const newBookAdd = { ...bookAdd };
                  newBookAdd.book.isbn10 = e.target.value;
                  setBookAdd(newBookAdd);
                }}
              />
              <br></br>
            </div>
            <div className="form-group">
              &nbsp;&nbsp; <i className="fas fa-barcode"></i> &nbsp;&nbsp;{" "}
              <label htmlFor="info">ISBN 13</label>
              <input
                type="text"
                className="form-control"
                id="ISBN-13"
                aria-describedby="ISBN"
                placeholder="Enter ISBN 13"
                onChange={(e) => {
                  const newBookAdd = { ...bookAdd };
                  newBookAdd.book.isbn13 = e.target.value;
                  setBookAdd(newBookAdd);
                }}
              />
              <br></br>
            </div>
            &nbsp;&nbsp; <i className="far fa-calendar-alt"></i> &nbsp;&nbsp;{" "}
            <label htmlFor="info">Published Date</label>
            <br></br>
            <input
              type="date"
              id="birthday"
              name="birthday"
              onChange={(e) => {
                const newBookAdd = { ...bookAdd };
                newBookAdd.book.publishedDate = e.target.value;
                setBookAdd(newBookAdd);
              }}
            />
            <br></br>
            <i className="fas fa-file-image"></i>&nbsp;&nbsp;
            <label htmlFor="img">Add a Book Cover</label>
            <p>You can either choose to add an image file or a URL</p>
            <i className="fas fa-file-image"></i>&nbsp;&nbsp;
            <input type="file" id="img" name="img" accept="image/*" />
            <br></br>
            <br></br>
            <p> Or </p>
            <i className="fas fa-link"></i>&nbsp;&nbsp;{" "}
            <label htmlFor="info">Book Cover URL</label>
            <input
              type="url"
              className="form-control"
              id="url"
              aria-describedby="url"
              placeholder="Enter a URl for the cover image for your book"
              onChange={(e) => {
                const newBookAdd = { ...bookAdd };
                newBookAdd.book.coverImageUrl = e.target.value;
                setBookAdd(newBookAdd);
              }}
            />
            <br></br>
            <p style={{ textalign: "left", marginBottom: "6%" }}>
              <span style={{ float: "right" }}>
                <button
                  type="addbook"
                  className="btn btn-primary"
                  onClick={handleAddBook}
                >
                  Add Book
                </button>
              </span>
            </p>
          </form>
        </div>
      </div>
    </div>
  );
};

const mapStateToProps = (state) => {
  return {
    id: state.security.user.id,
  };
};
export default connect(mapStateToProps, { addBook })(AddBook);

export const AddBookPlain = () => {
  return (
    <div data-testid="row" class="row">
      <div data-testid="container" class="container">
        <ToastContainer position="top-center" autoClose={2000} />

        <div class="column">
          <h1 data-testid="form-title">Sell A Book Form</h1>
          <h5 data-testid="form-description">
            Please fill in the form below with the details of the book you want
            to sell. Thank you for choosing our platform to sell your book!
          </h5>
          <hr></hr>
          <br></br>
          <form>
            <div class="form-group">
              <i class="fas fa-book-open"> &nbsp;&nbsp;</i>
              <label data-testid="book-title" for="Book Title">
                {" "}
                Book Title
              </label>
              <input
                type="text"
                class="form-control"
                id="Book Title"
                aria-describedby="Book Title"
                placeholder="Enter Book Title"
              />
            </div>
            <div class="form-group">
              <i class="fas fa-user"></i> &nbsp;&nbsp;{" "}
              <label for="author">Author</label>
              <input
                type="text"
                class="form-control"
                id="Author"
                aria-describedby="Author"
                placeholder="Enter Author Name"
              />
            </div>
            <div className="form-group">
              <i class="fa fa-list-alt" aria-hidden="true"></i>&nbsp;&nbsp;
              <label for="category">Category</label>
              <select
                name="categorytype"
                className="form-control form-control-md"
              >
                <option selected value="Science Fiction">
                  {" "}
                  Science Fiction
                </option>
                <option value="Novel">Novel</option>
                <option value="Fantasy">Fantasy</option>
                <option value="Thriller">Thriller</option>
                <option value="Romance">Romance</option>
              </select>
            </div>
            <div class="form-group">
              <i class="fas fa-info-circle"></i> &nbsp;&nbsp;{" "}
              <label for="status">Status (Published or Not)</label>
              <select className="form-control form-control-md">
                {" "}
                <option selected value="USED">
                  {" "}
                  Used
                </option>
                <option value="NEW">New</option>
              </select>
            </div>

            <div class="form-group">
              <i class="fas fa-dollar-sign"></i> &nbsp;&nbsp;{" "}
              <label for="price">Price</label>
              <input
                type="text"
                class="form-control"
                id="price"
                aria-describedby="price"
                placeholder="Enter Price"
              />
            </div>
            <div class="form-group">
              <i class="fas fa-info-circle"></i> &nbsp;&nbsp;{" "}
              <label for="quantity">Quantity</label>
              <input
                type="text"
                class="form-control"
                id="quantity"
                aria-describedby="quantity"
                placeholder="Enter Quantity"
              />
            </div>
            <div class="form-group">
              <i class="fas fa-info-circle"></i> &nbsp;&nbsp;{" "}
              <label for="quantity">Publisher</label>
              <input
                type="text"
                class="form-control"
                id="publisher-name"
                aria-describedby="publisher-name"
                placeholder="Enter Publisher"
              />
            </div>
            <div class="form-group">
              <i class="fas fa-bars"></i> &nbsp;&nbsp;{" "}
              <label for="long description">Long Description</label>
              <textarea
                class="form-control"
                id="long description"
                aria-describedby="long description"
                placeholder="Enter a long description"
              ></textarea>
            </div>

            <div class="form-group">
              <i class="fas fa-bars"></i> &nbsp;&nbsp;{" "}
              <label for="short description">Short Description</label>
              <textarea
                class="form-control"
                id="short description"
                aria-describedby="short description"
                placeholder="Enter a short description"
              ></textarea>
            </div>

            <div class="form-group">
              &nbsp;&nbsp; <i class="fas fa-barcode"></i> &nbsp;&nbsp;{" "}
              <label for="info">ISBN 10</label>
              <input
                type="text"
                class="form-control"
                id="ISBN-10"
                aria-describedby="ISBN"
                placeholder="Enter ISBN 10"
              />
              <br></br>
            </div>

            <div class="form-group">
              &nbsp;&nbsp; <i class="fas fa-barcode"></i> &nbsp;&nbsp;{" "}
              <label for="info">ISBN 13</label>
              <input
                type="text"
                class="form-control"
                id="ISBN-13"
                aria-describedby="ISBN"
                placeholder="Enter ISBN 13"
              />
              <br></br>
            </div>

            <form action="/action_page.php">
              &nbsp;&nbsp; <i class="far fa-calendar-alt"></i> &nbsp;&nbsp;{" "}
              <label for="info">Published Date</label>
              <br></br>
              <input type="date" id="birthday" name="birthday" />
            </form>

            <br></br>

            <form action="/action_page.php">
              <i class="fas fa-file-image"></i>&nbsp;&nbsp;
              <label for="img">Add a Book Cover</label>
              <p>You can either choose to add an image file or a URL</p>
              <i class="fas fa-file-image"></i>&nbsp;&nbsp;
              <input type="file" id="img" name="img" accept="image/*" />
              <br></br>
              <br></br>
              <p> Or </p>
              <i class="fas fa-link"></i>&nbsp;&nbsp;{" "}
              <label for="info">Book Cover URL</label>
              <input
                type="url"
                class="form-control"
                id="url"
                aria-describedby="url"
                placeholder="Enter a URl for the cover image for your book"
              />
            </form>
            <br></br>
            <p style={{ textalign: "left" }}>
              <span style={{ float: "right" }}>
                <button type="addbook" class="btn btn-primary">
                  Add Book
                </button>
                <p class="ex1"></p>
              </span>
            </p>
          </form>
        </div>
      </div>
    </div>
  );
};
