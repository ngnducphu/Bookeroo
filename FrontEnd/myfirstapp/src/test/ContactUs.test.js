import React from "react";
import { render, cleanup } from "@testing-library/react";
import { HeaderPlain } from "../components/Layout/Header";
import { ContactusPlain } from "../components/Pages/ContactUs";
import { BrowserRouter } from "react-router-dom";
import "@testing-library/jest-dom";
import renderer from "react-test-renderer";

afterEach(cleanup);

test("Render without crashing", () => {
  render(
    <BrowserRouter>
      <HeaderPlain />
    </BrowserRouter>
  );
});

test("renders  description ", () => {
  const { getByTestId } = render(
    <BrowserRouter>
      <ContactusPlain />
    </BrowserRouter>
  );
  const DescriptionElement = getByTestId("description");
  expect(DescriptionElement).toBeInTheDocument();
});

test("matches snapshot", () => {
  const tree = renderer
    .create(
      <BrowserRouter>
        <HeaderPlain />
      </BrowserRouter>
    )
    .toJSON();

  expect(tree).toMatchSnapshot();
});
