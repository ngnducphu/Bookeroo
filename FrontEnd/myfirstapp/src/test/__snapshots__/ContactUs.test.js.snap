// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<div>
  <nav
    className="navbar navbar-expand-sm navbar-dark bg-primary mb-4"
  >
    <div
      className="container"
    >
      <a
        className="navbar-brand"
        href="/main"
        onClick={[Function]}
      >
        Bookeroo
      </a>
      <button
        className="navbar-toggler"
        data-target="#mobile-nav"
        data-toggle="collapse"
        type="button"
      >
        <span
          className="navbar-toggler-icon"
        />
      </button>
      <div
        className="collapse navbar-collapse"
        id="mobile-nav"
      >
        <ul
          className="navbar-nav mr-auto"
        >
          <li
            className="nav-item"
          >
            <a
              className="nav-link"
              href="/aboutus"
              onClick={[Function]}
            >
              About us
            </a>
          </li>
          <li
            className="nav-item"
          >
            <a
              className="nav-link"
              href="/contactus"
              onClick={[Function]}
            >
              Contact us
            </a>
          </li>
          
        </ul>
        <ul
          className="navbar-nav ml-auto"
        >
          <li
            className="nav-item"
          >
            <a
              className="nav-link "
              href="/profile"
              onClick={[Function]}
            >
              My Profile
            </a>
          </li>
        </ul>
      </div>
    </div>
  </nav>
</div>
`;
